src/WebJobs.Extensions.DurableTask/Auth/AzureCredentialFactory.cs

```diff
@@ -72,6 +72,10 @@ public AppAuthTokenCredential Create(IConfiguration configuration, TimeSpan toke
                 RefreshOffset = tokenRefreshOffset,
             };
 
+            // The token credential will make background callbacks to renew the token.
+            // We suppress async flow to avoid logging scope from being captured as we do not know
+            // where this will be called from first.
+            using AsyncFlowControl flowControl = System.Threading.ExecutionContext.SuppressFlow();
```

Comment:
What logging scope is currently being captured, and why does it need to be suppressed? I'm missing a bit of background context on this. Just wondering how confident we feel that this fix will be effective.

Reply:
We use `ILogger` in the trace helper here. The log scope of whatever was present before calling `.Create` is captured.

I did manual tests with an `AsyncLocal` and verified before the value set outside is present during renew, and after the change it is no longer present.

Reply:
Makes sense at a high level. The thing I can't wrap my head around is that the thing we captured appears to have a lifetime that's different from this AzureCredentialFactory (a singleton). How is that possible? Shouldn't singletons only capture context that also has a singleton lifecycle? For this particular crash, it appears that a singleton object captured a per-request object, which seems very concerning to me.

Reply:
Yes, a singleton did capture a per-request object! But it didn't capture it through DI (nor at time of construction). It captured it through execution context and at the time/where `.Create` was called. Async locals are very powerful, but also lead to subtle issues like this.

So yes, it is a singleton, but the 'scope' I talk about in the PR is not related to service scope directly.

Reply:
Thanks for the clarification. I think I understand it now. Makes me much more wary of async locals as silent app killers. 😅